### PR TITLE
make deemphasis text accessible, add arg types to EzTextStyle and EzLabel

### PR DESCRIPTION
**Pull request by danidewitt**
## What did we change?
- Make deemphasis text accessible, this affects `EzTextStyle`, `EzLabel`, and `EzTable` column headers
- Add arg types to `EzLabel` and `EzTextStyle` (`EzTable` was already done)

## Why are we doing this?
- Accessibility

## Screenshot(s) / Gif(s):
Screenshot 2023-09-18 at 4 51 20 PM
Screenshot 2023-09-18 at 4 51 46 PM
Screenshot 2023-09-18 at 4 52 16 PM

## Checklist

- [ ] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes
 
[Contributing guidelines](https://recipe.ezcater.com/?path=/docs/guides-contributing--docs)

### CODE REVIEW

**Review by pgrippi — DISMISSED** [review]

File: packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx [hunk 1]
```diff
@@ -11,5 +61,21 @@ export default meta;
...
+    playroom: {
+      code: dedent`
+        <div>
+          <EzLabel use="primary" htmlFor="primary-input">Label</EzLabel>
+          <input id="primary-input" placeholder="Some text" />
+        </div>
+      `,
+    },
+  },
+  render: (args: Props) => (
```
Comment (noranda) [inline packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx:75]: I've been renaming these where needed to make the exports more explicit.

```suggestion
  render: (args: EzLabelProps) => (
```

File: packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx [hunk 2]
```diff
@@ -1,8 +1,58 @@
...
+  argTypes: {
+    as: {
+      control: {type: 'select'},
+      description: 'The element type for the label.',
+      options: ['label', 'legend', 'div'],
+      table: {
+        defaultValue: {summary: 'label'},
+        type: {summary: 'label | legend | div'},
+      },
+      type: {name: 'string', required: false},
```
Comment (noranda) [inline packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx:16]: This line can be removed (same with the other props). I still need to go back and clean up some of the other components, but it's only needed if the prop is required.
```suggestion
```

File: packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx [hunk 3]
```diff
@@ -1,8 +1,58 @@
...
+    htmlFor: {
+      control: {type: 'text'},
+      description:
+        'Associates a label with an input, and should be the same as the id of the input. Alternatively, nest the input element directly inside the label.',
+      table: {
+        type: {summary: 'string'},
+      },
+      type: {name: 'string', required: false},
+    },
+    use: {
```
Comment (noranda) [inline packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx:27]: I've been putting these props in alphabetical order, just to make it easier to find props for components with a ton of them. The table will sort them, but it helps maintenance.

File: packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx [hunk 4]
```diff
@@ -1,8 +1,58 @@
...
+      control: {type: 'select'},
+      description: 'The style of the label for the selected size.',
```
Comment (noranda) [inline packages/recipe/src/components/EzLabel/Documentation/Stories/Default.stories.tsx:29]:
```suggestion
      description: 'The use or variant of the label.',
```
To be consistent with descriptions of other components with this prop.

File: packages/recipe/src/components/EzTextStyle/Documentation/Stories/Default.stories.tsx [hunk 5]
```diff
@@ -1,8 +1,31 @@
...
+      options: ['center', 'left', 'right'],
+      table: {
+        defaultValue: {summary: 'left'},
+        type: {summary: 'center | left | right'},
+      },
+      type: {name: 'string', required: false},
+    },
+    use: {
+      control: {type: 'select'},
+      description: 'The emphasis style of the text element.',
```
Comment (noranda) [inline packages/recipe/src/components/EzTextStyle/Documentation/Stories/Default.stories.tsx:21]:
```suggestion
      description: 'The use or variant of the text element.',
```

File: packages/recipe/src/components/EzTextStyle/Documentation/Stories/Default.stories.tsx [hunk 6]
```diff
@@ -11,5 +34,13 @@ export default meta;
...
 export const Default: Story = {
-  render: () => <div>Coming soon...</div>,
+  args: {},
+  parameters: {
+    playroom: {
+      code: dedent`
+        <EzTextStyle>This is some example text.</EzTextStyle>
+      `,
+    },
+  },
```
Comment (noranda) [inline packages/recipe/src/components/EzTextStyle/Documentation/Stories/Default.stories.tsx:44]:
```suggestion
    playroom: {code: <EzTextStyle>This is some example text.</EzTextStyle>},
```
No `dedent` needed here. It's only needed if there is more than one line.

**Review by noranda — CHANGES_REQUESTED** [review]
Just a few minor change requests (apply them to both components). Otherwise, looks good!

**Comment by noranda** [comment]
We also might want to take a closer look at `EzField` dropdowns with groups after this change. The new color makes the group labels almost indistinguishable from the options. @alexanderlippner16?


Screenshot 2023-09-19 at 8 58 58 AM

**Comment by danidewitt** [comment]
@noranda i added a horizontal line between the grouping label and its items per @alexanderlippner16 request
Screenshot 2023-09-25 at 3 41 46 PM

File: packages/recipe/src/components/EzField/EzListBox.tsx [hunk 7]
```diff
@@ -60,6 +61,17 @@ const disabledOption = theme.css({
   opacity: 0.6,
 });
 
+const groupDivider = theme.css({
+  '&': {
+    color: '#F0F2F7', // this is a new color from color tokens, we're adding this divider for accessibility purposes before moving to new color tokens
```
Comment (pgrippi) [inline packages/recipe/src/components/EzField/EzListBox.tsx:66]: minor but can this `color` rule be moved up next to all of the other ones? I don't think we need to nest it under `&`

**Review by pgrippi — APPROVED** [review]

File: packages/recipe/src/components/EzField/EzListBox.tsx (hunk 7 above)
Comment (danidewitt) [reply packages/recipe/src/components/EzField/EzListBox.tsx:66]: it's under `&` for specificity to avoid using !important

**Review by noranda — APPROVED** [review]

**Pull request merged**